src/main/java/com/khutircraftubackend/category/CategoryController.java

```diff
@@ -1,71 +1,78 @@
 
 import com.khutircraftubackend.category.request.CategoryCreateRequest;
-import com.khutircraftubackend.category.response.CategoryResponse;
 import com.khutircraftubackend.category.request.CategoryUpdateRequest;
+import com.khutircraftubackend.category.response.CategoryResponse;
 import lombok.RequiredArgsConstructor;
 import lombok.extern.slf4j.Slf4j;
 import org.springframework.http.HttpStatus;
-import org.springframework.http.ResponseEntity;
 import org.springframework.security.access.prepost.PreAuthorize;
 import org.springframework.web.bind.annotation.*;
 import org.springframework.web.multipart.MultipartFile;
 
 import java.io.IOException;
+import java.util.Collection;
 import java.util.List;
 
 @RestController
 @RequestMapping("/v1/category")
 @RequiredArgsConstructor
 @Slf4j
 public class CategoryController {
-
-    private final CategoryService categoryService;
-    private final CategoryMapper categoryMapper;
-
-    @GetMapping("/")
-    public List<CategoryResponse> getAllRootCategories() {
-        return categoryService.getAllRootCategories();
-    }
-
-    @GetMapping("/parent-id/{id}")
-    public List<CategoryResponse> getAllCategoriesByParentId(
-            @PathVariable Long id) {
-        return categoryService.getAllByParentCategoryId(id);
-    }
-
-    @PostMapping("/")
-    @PreAuthorize("hasRole('ADMIN')")
-    @ResponseStatus(HttpStatus.OK)
-    public CategoryResponse createCategory(
-            @ModelAttribute CategoryCreateRequest request,
-            @RequestPart(value = "iconFile", required = false) MultipartFile iconFile) throws IOException {
-
-        CategoryEntity category = categoryService.createCategory(request, iconFile);
-
-        return categoryMapper.toCategoryResponse(category);
-    }
-
-    @PatchMapping("/{id}")
-    @PreAuthorize("hasRole('ADMIN')")
-    @ResponseStatus(HttpStatus.OK)
-    public CategoryResponse updateCategory(
-            @PathVariable Long id,
-            @ModelAttribute CategoryUpdateRequest request,
-            @RequestPart(value = "iconFile", required = false) MultipartFile iconFile) throws IOException {
-
-        CategoryEntity updateCategory = categoryService.updateCategory(id, request, iconFile);
-
-        return categoryMapper.toCategoryResponse(updateCategory);
-    }
-
-    @DeleteMapping("/{id}")
-    @PreAuthorize("hasRole('ADMIN')")
-    @ResponseStatus(HttpStatus.NO_CONTENT)
-    public ResponseEntity<Void> deleteCategory(@PathVariable Long id,
-                                               @RequestParam(required = false) boolean forceDelete) {
-        categoryService.deleteCategory(id, forceDelete);
-        return ResponseEntity.noContent().build();
-    }
-
+	
+	private final CategoryService categoryService;
+	private final CategoryMapper categoryMapper;
+	
+	@GetMapping("/")
+	public Collection<CategoryResponse> getAllRootCategories() {
+		
+		List<CategoryEntity> categoryEntity = categoryService.getAllRootCategories();
```

Comment:
if it's a "collection", then is must be plural: `categoryEntities` :)

Reply:
and it's more generic type now: `Collection<CategoryEntity>`

Reply:
though `categoryService.getAllRootCategories()` returns a List, but it's a subclass of `Collection`.
so, we could have here: `Collection<CategoryEntity> categoryEntities` 
similar to the the method below. it's absolutely ok.